index.js

```diff
@@ -400,7 +396,7 @@ function Physics (mcData, world) {
       vel.x *= horizontalInertia
       vel.z *= horizontalInertia
 
-      if (entity.isCollidedHorizontally && doesNotCollide(world, pos.offset(vel.x, vel.y + 0.6 - pos.y + lastY, vel.z))) {
+      if (doesNotCollide(world, pos.offset(vel.x, vel.y + 0.6 - pos.y + lastY, vel.z))) {
```

Comment:
Line 274 of https://www.mcpk.wiki/wiki/SourceCode:EntityLivingBase

I'm not excluding that the sources we find online are a bit old and that the physic code of vanilla may have evolved. But this should be double checked.

Reply:
Im not sure whether this is true to the vanilla behaviour, however the impulse isin't applied when checking isCollidedHorizontally (or vertically) because they are only active at time of collision.

Reply:
Its also peculiar why would the check be checking horizontal collision rather than vertical.

Reply:
This code is when you are on the border of a pool, if you collide with a solid block it boost you a bit so you can more easily walk out of water.

An other source: https://github.com/eldariamc/client/blob/master/src/main/java/net/minecraft/entity/EntityLivingBase.java#L1574

Reply:
I see, so this change must be reverted.